src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -272,7 +272,9 @@ private async Task ProcessWebConsoleLinesQueueAsync(bool runOnce = false)
                     linesCounter++;
 
                     // process at most about 500 lines of web console line during regular timer dequeue task.
-                    if (!runOnce && linesCounter > 500)
+                    // Send the first line of output to the customer right away
+                    // It might take a while to reach 500 line outputs, which would cause delays before customers see the first line
+                    if ((!runOnce && linesCounter > 500) || _firstConsoleOutputs)
```

Comment:
does this matter? it will break out the loop when there is nothing in the queue or reach 500 lines.